lnd_test.go

```diff
@@ -6260,14 +6282,9 @@ func testRevokedCloseRetributionZeroValueRemoteOutput(net *lntest.NetworkHarness
 
 	// Query the mempool for the breaching closing transaction, this should
 	// be broadcast by Carol when she force closes the channel above.
-	txid, err := waitForTxInMempool(net.Miner.Node, 20*time.Second)
-	if err != nil {
-		t.Fatalf("unable to find Carol's force close tx in mempool: %v",
-			err)
-	}
-	if *txid != *closeTxId {
-		t.Fatalf("expected closeTx(%v) in mempool, instead found %v",
-			closeTxId, txid)
+	if !waitForTxidInMempool(net.Miner.Node, closeTxId, 20*time.Second) {
```

Comment:
Will the sweep broadcast be deterministic? If that's the case I feel we should wait for two txes instead.

Reply:
It will. The flake has only seem to come up with these changes introduced. Should just be a timing issue - but I decided to go with this approach based on the comment above wanting to ensure that the transaction was broadcast.

Reply:
Why not wait for both txes then?

Reply:
This wasn't needed after all. The counterparty shouldn't act upon mempool spends. Not sure how this happened, but haven't been able to reproduce it after many runs locally and on Travis.

Reply:
Turns out the flake I was seeing has already been addressed in https://github.com/lightningnetwork/lnd/pull/1621/commits/90f9fa2c14105cf06eae326a09d845098b178e4e#diff-61b0847f8902e7a5d746ba18a02649d5R12292.